src/middlewared/middlewared/pytest/unit/test_validators.py

```diff
@@ -37,3 +40,32 @@ def test__email_schema(email, should_raise):
     else:
         with pytest.raises(ValueError):
             Email()(email)
+
+
+@pytest.fixture(scope="function")
+def setup_mnt(tmpdir):
+    os.makedirs('/mnt/pool/foo')
+    try:
+        os.symlink(tmpdir, '/mnt/pool/symlink')
+        yield
+    finally:
+        shutil.rmtree('/mnt/pool')
+
+
+@pytest.mark.parametrize("path,should_raise", [
+    ('/tmp', True),
+    ('EXTERNAL://smb_server.local/SHARE', True),
+    ('/mnt/does_not_exist', True),
+    ('/mnt/pool/foo', False),
+    ('/mnt/pool', False),
+    ('/mnt/pool//symlink', True)
```

Comment:
Are the double slash intended here? (`//`)  Suspect not.

Also maybe add another test where the symlink /tmpdir and points under /mnt/foo